packages/host/app/components/operator-mode/code-submode/playground/playground-content.gts

```diff
@@ -199,6 +201,36 @@ export default class PlaygroundContent extends Component<Signature> {
   @tracked private fieldChooserIsOpen = false;
   @tracked private cardResource: ReturnType<getCard> | undefined;
 
+  private get specQuery(): Query {
+    return {
+      filter: {
+        on: specRef,
+        eq: { ref: this.args.codeRef },
+      },
+    };
+  }
+
+  private searchFieldSpec = getSearch(
+    this,
+    () => this.specQuery,
+    () => this.realmServer.availableRealmURLs,
+    { isLive: true, isAutoSaved: true },
+  );
+
+  private get canGenerateFieldSpec() {
+    if (
+      !this.args.isFieldDef ||
+      !this.canWriteRealm ||
+      this.createNewCard.isRunning
+    ) {
+      return false;
+    }
+    return (
+      !this.searchFieldSpec.isLoading &&
+      this.searchFieldSpec.instances?.length === 0
+    );
+  }
```

Comment:
this checks if there's a preexisting spec as well as other conditions for running createNew to autogenerate a new spec